### DISCUSSION TITLE
reinstall/repair broken program

### DISCUSSION THREAD

**Discussion by honzanugeta** [ChrisTitusTech/winutil#1978] [Ideas]
Could a repair/reinstall feature be added ? like when a user wants to update all his programs and winutil finds one that is broken. it automatically (or asks to) fixes it or reinstalls it.

**Comment by Marterich** [comment]
I'm actually not sure if this is actually possible to implement. The main problem will become to detect when an application is "broken" or "doesn't work properly". I think the smartest way to go about working with broken programs at the moment is to simply select it, uninstall it and then reinstall it afterwards. 

There isn't an easy "fix all" solution for anything in Windows that actually properly works and it almost every time requires at least some kind of manual intervention to fix problems.

I think the most that winutil could do here is check if the update was successful and if not, try reinstalling the program, but i'm actually not sure if this would result in a different experience from simply running winget upgrade ...

**Reply by generalbarca69** [reply]
Good day, I used the chris titus noattend xml. Specifically, i flashed win11 and win 10 iso to two different usb using rufus, and I made sure to uncheck all the instructed options, lastly i copy the xml into the root of the usb. win install was successful, but seems the win 11 and win 10 (both OS) virus and threat protection keeps crashing. any suggestions how to fix it?
